net/mptcp/mptcp_ecf.c

```diff
@@ -0,0 +1,384 @@
+// SPDX-License-Identifier: GPL-2.0
+/*	MPTCP ECF Scheduler
+ *
+ *	Algorithm Design:
+ *	Yeon-sup Lim <ylim@cs.umass.edu>
+ *	Don Towsley <towsley@cs.umass.edu>
+ *	Erich M. Nahum <nahum@us.ibm.com>
+ *	Richard J. Gibbens <richard.gibbens@cl.cam.ac.uk>
+ *
+ *	Initial Implementation:
+ *	Yeon-sup Lim <ylim@cs.umass.edu>
+ *
+ *	Additional Authors:
+ *	Daniel Weber <weberd@cs.uni-bonn.de>
+ *
+ *	This program is free software; you can redistribute it and/or
+ *	modify it under the terms of the GNU General Public License
+ *	as published by the Free Software Foundation; either version
+ *	2 of the License, or (at your option) any later version.
+ */
+
+#include <linux/module.h>
+#include <net/mptcp.h>
+#include <trace/events/tcp.h>
+
+static unsigned int r_beta __read_mostly = 4; /* beta = 1/r_beta = 0.25 */
+module_param(r_beta, int, 0644);
+MODULE_PARM_DESC(r_beta, "beta for ECF");
```

Comment:
Ok, I will change this. But the same also applies to the module parameters of the BLEST an RR scheduler as well.

Reply:
Good point. Do not hesitate to fix that for the other schedulers.

@cpaasch : or do you think it is not needed?

Reply:
@matttbe - I think that would be good to change.
@s6dlwebe - Let us know if you want to do that change in a separate PR. Otherwise, I will add it to the generic todo-list.

Reply:
> @matttbe - I think that would be good to change.
> @s6dlwebe - Let us know if you want to do that change in a separate PR. Otherwise, I will add it to the generic todo-list.

I just updated the PR onto mptcp_trunk and added the `mptcp_ecf_` prefix to the module parameter. For now, all schedulers are written in the same way. As @matttbe already pointed out, all schedulers use a mix of names for internal structures and static functions that we should change.

@cpaasch - I would prefer a separate PR for cleaning up all the scheduler naming mixup and add a todo to reduce the code duplication later. This will make it easier to keep track of the name changes and refactoring.

Reply:
Sounds good to me!

Reply:
and thank you for the update! :)